lib/Doctrine/Common/NotifyPropertyChanged.php

```diff
@@ -2,19 +2,24 @@
 
 namespace Doctrine\Common;
 
-/**
- * Interface for classes that notify event listeners of changes to their managed properties.
- *
- * This interface is implemented by objects that manually want to notify their object manager or
- * other listeners when properties change, instead of relying on the object manager to compute
- * property changes itself when changes are to be persisted.
- */
-interface NotifyPropertyChanged
-{
+if (!interface_exists(\Doctrine\Persistence\NotifyPropertyChanged::class, false)) {
+    @trigger_error(
+        'The '.__NAMESPACE__.'\NotifyPropertyChanged class is deprecated since doctrine/persistence 1.2 and will be removed in 2.0.'
+        .' Use \Doctrine\Persistence\NotifyPropertyChanged instead.',
+        E_USER_DEPRECATED
+    );
+}
```

Comment:
Are you ok with this or do you think line 20 should be enough?

Reply:
If the DebugClassLoader already triggers a deprecation based on the `@deprecated` annotation we should skip this to avoid duplicate deprecation messages. Otherwise, I definitely want to keep these in.

// cc @nicolas-grekas

Reply:
That practice on Symfony is to still add all possible runtime deprecations, because using the DebugClassLoader is optional, especially when using packages standalone.

Reply:
That clears it up. Thanks Nicolas!

@greg0ire I would suggest we add this deprecation message to all deprecated classes/interfaces then.